src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
I think it's better to convert to bool/true/false a variable/retval at a time; both the declatation, all assignments and all checks at once. NOTDONE is very rare, only two retvals in `eval.c` and one in `edit.c`. Converting other retvals to bool  makes it clear that there really only are two possible outcomes; it could simply be a convention that `true` is success and `false` is failure. (Or alternatively _another_ two-value enum, but I think bool is clear enough)